src/chronify/time_series_mapper_base.py

```diff
@@ -1,5 +1,100 @@
 import abc
+import logging
+from functools import reduce
+from operator import and_
+
+import pandas as pd
+from sqlalchemy import Engine, MetaData, Table, select, text
+
+from chronify.sqlalchemy.functions import write_database
+from chronify.models import TableSchema
+from chronify.exceptions import TableAlreadyExists
+from chronify.utils.sqlalchemy_table import create_table
+from chronify.time_series_checker import check_timestamps
+from chronify.time_configs import DatetimeRange
+
+logger = logging.getLogger(__name__)
 
 
 class TimeSeriesMapperBase(abc.ABC):
     """Maps time series data from one configuration to another."""
+
+
+def apply_mapping(
+    df_mapping: pd.DataFrame,
+    mapping_table_name: str,
+    from_schema: TableSchema,
+    to_schema: TableSchema,
+    engine: Engine,
+    metadata: MetaData,
+) -> None:
+    """
+    Apply mapping to create result table with process to clean up and roll back if checks fail
+    """
+    if mapping_table_name in metadata.tables:
+        msg = (
+            f"table {mapping_table_name} already exists, delete it or use a different table name."
+        )
+        raise TableAlreadyExists(msg)
+
+    time_configs = [to_schema.time_config]
+    if isinstance(from_schema.time_config, DatetimeRange):
+        from_time_config = from_schema.time_config.model_copy()
+        from_time_config.time_column = "from_" + from_time_config.time_column
```

Comment:
Open to suggestion as I don't love this implementation either.

`df_mapping` creation is in the child mapper classes where it is clear there's a from_timestamp in DatetimeRange. Also 

Option 1: I can make configs as an additional input to `apply_mapping()` but it feels out of place since one of the config is already in to_schema.

Option 2: Make apply_mapping() in each child class. This feels like a lot of duplicated code as the extra config is the only additional input required for DatetimeRange mapping.

Option 3: Suggestion

Reply:
The fundamental issue is that this time table does not fit in the definition of `TableSchema`, which only allows one time column. Another option is to create a new schema which allows multiple time columns. It could be something much more generic because it is not user facing. Something like
```
class TimeTableSchema(ChronifyBaseModel):
    """Defines the schema for a time series table stored in the database."""

    name: Annotated[
        str,
        Field(description="Name of the table or view in the database.", frozen=True),
    ]
    time_config: list[TimeConfig]
    other_columns: list[str]
```